kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/configuration/KafkaErrorHandler.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.homeoffice.digital.sas.kafka.configuration;
+
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_STOPPING_CONSUMING;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.listener.KafkaListenerErrorHandler;
+import org.springframework.stereotype.Component;
+import uk.gov.homeoffice.digital.sas.kafka.exceptions.KafkaConsumerException;
+
+@Component
+@Slf4j
+public class KafkaErrorHandler {
```

Comment:
Is this a config or a component. let's discuss this offline

Reply:
This has been moved to an actual handler instead of config class.